includes/class-woocommerce-delivery-notes.php

```diff
@@ -535,13 +535,24 @@ public function is_woocommerce_activated() {
 		 */
 		public function ts_reset_tracking_setting() {
 
-			if ( isset( $_GET ['ts_action'] ) && 'reset_tracking' === $_GET ['ts_action'] ) { //phpcs:ignore
-				if ( is_user_logged_in() && current_user_can( 'manage_options' ) ) {
-					Tyche_Plugin_Tracking::reset_tracker_setting( 'wcdn' );
-					$ts_url = remove_query_arg( 'ts_action' );
-					wp_safe_redirect( $ts_url );
-				}
+			$nonce = $_POST['ts_tracker_nonce'];//phpcs:ignore
+			if ( ! wp_verify_nonce( $nonce, 'tracking_notice' ) ) {
+				return;
+			}
+			if ( isset( $_POST['plugin_short_name'] ) ) { //phpcs:ignore
+				$plugin_short_name = $_POST['plugin_short_name']; //phpcs:ignore
 			}
+
+			delete_option( $plugin_short_name . '_allow_tracking' );
```

Comment:
@nidhitatosaniya The options data are already being deleted in the plugin_tracking file. Ideally, we should check here if the option still exists before going ahead to delete it.